presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -443,32 +440,6 @@ else if (firstDependencies.stream().allMatch(right::canResolve) && secondDepende
         return new RelationPlan(root, analysis.getScope(node), outputs);
     }
 
-    private void addNullFilters(List<Expression> conditions, Join.Type joinType, Expression left, Expression right)
```

Comment:
We discussed this [earlier in this thread](https://github.com/prestodb/presto/pull/18938#discussion_r1090182593). The consensus was to keep a single feature flag, which necessitates removing the older code. Note that this PR does completely cover functionality that was added prior, so to me, it makes sense to remove the older code with this PR

Reply:
If you want to remove the old code within the same PR, please split it into two commits then. This will help make the change clean and easier to review.